application/libraries/PluginManager/PluginManager.php

```diff
@@ -634,17 +638,38 @@ protected function getPluginName(string $class, \ExtensionConfig $extensionConfi
     }
 
     /**
+     * Returns true if the plugin name is whitelisted or the whitelist is disabled.
      * @param string $pluginName
      * @return boolean
      */
-    private function _checkWhitelist($pluginName)
+    public function isWhitelisted($pluginName)
     {
         if (App()->getConfig('usePluginWhitelist')) {
             $whiteList = App()->getConfig('pluginWhitelist');
-            $coreList = App()->getConfig('pluginCoreList');
+            $coreList = self::getPluginCoreList();
             $allowedPlugins =  array_merge($coreList, $whiteList);
             return array_search($pluginName, $allowedPlugins) !== false;
         }
         return true;
     }
+
+    /**
+     * Return the core plugin list
+     * No way to update by php or DB
+     * @return string[]
+     */
+    private static function getPluginCoreList()
+    {
+        return [
+            'AuditLog',
+            'ExportR',
+            'ExportSTATAxml',
+            'ExportSPSSsav',
+            'extendedStartPage',
+            'oldUrlCompat',
+            'AuthLDAP',
+            'Authdb',
+            'Authwebserver'
+        ];
```

Comment:
Lack of some core plugin : different ways to fix 

- don't allow some core plugins (maybe disallow Authwebserver for example)
- fix the list adding all core plugins
- get the list from directory (and cache it)

Reply:
Sorry for the list i give … (copy pasted from yours)

Reply:
Sorry, what do you mean by `Lack of some core plugin : different ways to fix` ?

Reply:
Lack of some plugins : https://github.com/LimeSurvey/LimeSurvey/tree/master/application/core/plugins 
All core plugins not included here :)

Reply:
Then the 3 point is 3 way to fix it :) Why allow AuthWebsrevr (that can broke access) and disallow expressionFixedDbVar ?

Reply:
In fact your 1st idea to have a config array for this is not a bad solution … but name it 'allowedCorePlugin' and always force AuthDB (currently)

Reply:
I am not sure about changing the name.
If this list is `allowedCorePlugins`, would there be a `notAllowedCorePlugins`? would there be any core plugin not allowed?

Reply:
I suggest that disabling core plugins should be another story.

Reply:
> I suggest that disabling core plugins should be another story.

Then you need to fix the list : expressionFixedDbVar, expressionQuestionHelp … etc … :).

By «fix the list adding all core plugins» or «get the list from directory (and cache it)»

Reply:
> fix the list adding all core plugin

Agree.